src/client/serialize/daos_serialize.c

```diff
@@ -1175,3 +1252,1807 @@ daos_cont_deserialize_attrs(char *filename, uint64_t *num_attrs,
+	if (status < 0) {
+		D_ERROR("Failed to insert record dataset ID\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+
+	status = H5Tinsert(akey_memtype, "Record Single Value",
+			   HOFFSET(struct akey, rec_single_val), akey_vtype);
+	if (status < 0) {
+		D_ERROR("Failed to insert record single value\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+
+	status = H5Tinsert(akey_memtype, "Akey Value", HOFFSET(struct akey, akey_val), akey_vtype);
+	if (status < 0) {
+		D_ERROR("Failed to insert akey value\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+
+	akey_dset = H5Dcreate(hdf5->file, "Akey Data", akey_memtype, akey_dspace, H5P_DEFAULT,
+			      H5P_DEFAULT, H5P_DEFAULT);
+	if (akey_dset < 0) {
+		D_ERROR("Failed to create akey dset\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+
+	status = H5Dwrite(akey_dset, akey_memtype, H5S_ALL, H5S_ALL, H5P_DEFAULT, *(hdf5->ak));
+	if (status < 0) {
+		D_ERROR("Failed to write akey dset\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+out:
+	if (akey_dspace >= 0)
+		H5Sclose(akey_dspace);
+	if (akey_memtype >= 0)
+		H5Tclose(akey_memtype);
+	if (akey_vtype >= 0)
+		H5Tclose(akey_vtype);
+	if (akey_dset >= 0)
+		H5Dclose(akey_dset);
+	return rc;
+}
+
+static int
+serialize_version(struct hdf5_args *hdf5, float version)
+{
+	int	rc = 0;
+	hid_t	status = 0;
+	char	*version_name = "Version";
+	hsize_t	version_attr_dims[1];
+	hid_t	version_attr_type = -1;
+	hid_t	version_attr_dspace = -1;
+	hid_t	version_attr = -1;
+
+	version_attr_dims[0] = 1;
+	version_attr_type = H5Tcopy(H5T_NATIVE_FLOAT);
+	status = H5Tset_size(version_attr_type, 4);
+	if (status < 0) {
+		D_ERROR("Failed to set version attribute size\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+	if (version_attr_type < 0) {
+		D_ERROR("Failed to create version attribute datatype\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+	version_attr_dspace = H5Screate_simple(1, version_attr_dims, NULL);
+	if (version_attr_dspace < 0) {
+		D_ERROR("Failed to create version attribute dataspace\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+	version_attr = H5Acreate2(hdf5->file, version_name, version_attr_type, version_attr_dspace,
+				  H5P_DEFAULT, H5P_DEFAULT);
+	if (version_attr < 0) {
+		D_ERROR("Failed to create version attribute\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+	status = H5Awrite(version_attr, version_attr_type, &version);
+	if (status < 0) {
+		D_ERROR("Failed to write attribute\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+out:
+	if (version_attr_type >= 0)
+		H5Tclose(version_attr_type);
+	if (version_attr_dspace >= 0)
+		H5Sclose(version_attr_dspace);
+	if (version_attr >= 0)
+		H5Aclose(version_attr);
+	return rc;
+}
+
+int
+daos_cont_serialize(daos_prop_t *props, int num_attrs, char **names, char **buffers, size_t *sizes,
+		    int *total_oids, int *total_dkeys, int *total_akeys, uint64_t *bytes_read,
+		    daos_handle_t coh, char *filename)
+{
+	int			rc = 0;
+	int			rc2 = 0;
+	hid_t			status = 0;
+	int			i = 0;
+	int			dkey_index = 0;
+	int			akey_index = 0;
+	struct hdf5_args	hdf5 = {0};
```

Comment:
probably shouldn't use a library name as the variable name.

Reply:
Renamed hdf5 -> args